### PR TITLE
TRUNK-6202 Replace Hibernate Criteria API with JPA for HibernateOrderSetDAO

### PR DESCRIPTION
**Pull request by k4pran**
## Description of what I changed

This task is to migrate the deprecated hibernate criteria to jpa criteria for the HibernateOrderSetDAO class.

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-6202

## Checklist: I completed these to help reviewers :)

- [X] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [X] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [X] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [X] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [X] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/test/java/org/openmrs/api/OrderSetServiceTest.java [hunk 1]
```diff
@@ -475,5 +475,32 @@ public void unretireOrderSetAttributeType_shouldUnretireARetiredOrderSetAttribut
...
 	}
 
+	/**
+	 * @see OrderSetService#getOrderSetAttributeTypeByName(String) 
+	 */
+	@Test
+	public void getOrderSetAttributeTypeByName_shouldGetMatchingOrderSetAttributeType() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+
+		String attributeTypeName = "Audit Date";
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/OrderSetServiceTest.java:485]: Shouldn't this be a constant?

File: api/src/test/java/org/openmrs/api/OrderSetServiceTest.java [hunk 2]
```diff
@@ -475,5 +475,32 @@ public void unretireOrderSetAttributeType_shouldUnretireARetiredOrderSetAttribut
...
+	}
 
+	/**
+	 * @see OrderSetService#getOrderSetAttributeTypeByName(String)
+	 */
+	@Test
+	public void getOrderSetAttributeTypeByName_shouldReturnNullForNonExistentName() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+
+		String attributeTypeName = "Non existent name";
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/OrderSetServiceTest.java:500]: Shouldn't this be a constant?

File: api/src/test/java/org/openmrs/api/OrderSetServiceTest.java [hunk 3]
```diff
@@ -475,5 +478,28 @@ public void unretireOrderSetAttributeType_shouldUnretireARetiredOrderSetAttribut
...
+	/**
+	 * @see OrderSetService#getOrderSetAttributeTypeByName(String)
+	 */
+	@Test
+	public void getOrderSetAttributeTypeByName_shouldReturnNullForNonExistentName() {
+		executeDataSet(ORDER_SET_ATTRIBUTES);
+
+		OrderSetAttributeType attributeType = orderSetService.getOrderSetAttributeTypeByName(INVALID_AUDIT_DATE);
+
+		assertNull(attributeType, "The fetched OrderSetAttributeType should not be null");
```
Comment (dkayiwa) [inline api/src/test/java/org/openmrs/api/OrderSetServiceTest.java:503]: Should not be or should be null?

File: api/src/test/java/org/openmrs/api/OrderSetServiceTest.java (hunk 1 above)
Comment (k4pran) [reply api/src/test/java/org/openmrs/api/OrderSetServiceTest.java]: Done

File: api/src/test/java/org/openmrs/api/OrderSetServiceTest.java (hunk 2 above)
Comment (k4pran) [reply api/src/test/java/org/openmrs/api/OrderSetServiceTest.java]: Done

File: api/src/test/java/org/openmrs/api/OrderSetServiceTest.java (hunk 3 above)
Comment (k4pran) [reply api/src/test/java/org/openmrs/api/OrderSetServiceTest.java]: Yes, fixed

**Comment by dkayiwa** [comment]
Thanks @k4pran 👍

**Pull request merged**